internal/k8s/controller.go

```diff
@@ -969,15 +987,49 @@ func (lbc *LoadBalancerController) syncVirtualServer(task task) {
 		lbc.recorder.Eventf(vsr, vsrEventType, vsrEventTitle, msg)
 
 		if lbc.reportVsVsrStatusEnabled() {
-			virtualServersForVSR := findVirtualServersForVirtualServerRoute(lbc.getVirtualServers(), vsr)
-			err = lbc.statusUpdater.UpdateVirtualServerRouteStatusWithReferencedBy(vsr, state, vsrEventTitle, msg, virtualServersForVSR)
+			vss := []*conf_v1.VirtualServer{vs}
+			err = lbc.statusUpdater.UpdateVirtualServerRouteStatusWithReferencedBy(vsr, state, vsrEventTitle, msg, vss)
 
 			if err != nil {
 				glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
 			}
 		}
+
+		handledVSRs = append(handledVSRs, vsr)
 	}
 
+	orphanedVSRs := filterOutVirtualServerRoutes(previousVSRs, handledVSRs)
+	reason := "NoVirtualServerFound"
+	for _, vsr := range orphanedVSRs {
+		msg := fmt.Sprintf("No VirtualServer references VirtualServerRoute %v/%v", vsr.Namespace, vsr.Name)
+		lbc.recorder.Eventf(vsr, api_v1.EventTypeWarning, "Ignored", msg)
+		if lbc.reportVsVsrStatusEnabled() {
+			var emptyVSS []*conf_v1.VirtualServer
+			err := lbc.statusUpdater.UpdateVirtualServerRouteStatusWithReferencedBy(vsr, conf_v1.StateInvalid, reason, msg, emptyVSS)
+			if err != nil {
+				glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
+			}
+		}
+	}
+}
+
+func filterOutVirtualServerRoutes(previousVSRs []*conf_v1.VirtualServerRoute, handledVSRs []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
+	var orphanedVSRs []*conf_v1.VirtualServerRoute
+	for _, r := range previousVSRs {
```

Comment:
better to not name variables like `r1`. 
name like `r1` doesn't provide any context. it makes the code less readable.

Consider the following names for the loop:

`for _, r := range previousVSRs` -> `for _, prev := range previousVSRs`
`for _, r1 := range handledVSRs` -> `for _, handled := range handledVSRs`

`key` - > `prevKey`
`vsrKey` -> `handledKey`